lib/resty/upstream/healthcheck.lua

```diff
@@ -513,10 +512,8 @@ local function preprocess_peers(peers)
         local p = peers[i]
         local name = p.name
         if name then
-            local idx = str_find(name, ":", 1, true)
-            if idx then
-                p.host = sub(name, 1, idx - 1)
-                p.port = tonumber(sub(name, idx + 1))
+            if name:sub(-1, -1)~=']' then
+                p.host, p.port = name:match'(.*):(.*)'
```

Comment:
This is less efficient than the original `string.find + string.sub` combination since `string.match` can not be JIT compiled yet.